### DISCUSSION TITLE
Generic part suggestions

### DISCUSSION THREAD

**Discussion by ataradov** [ataradov/usb-sniffer#2] [Announcements]
For EEPROM I just put 128 Kbit version to get full 16 KByte. But anything starting with 24LC32 would work.

Type-C connectors I use are https://www.lcsc.com/product-detail/_XUNPU-_C2835315.html . But any 12-pin connector of this style would work, they are pretty much standard. 

USB-A was designed for this https://www.lcsc.com/product-detail/_Shenzhen-Kinghelm-Elec-_C2979055.html connector, but I would not recommend it. It is flimsy and there is too much contact bounce. It does not matter for device applications, but on the sniffer you would see all that bounce in the log.

The footprint is very standard as well and would fit any connector of this style https://www.mouser.com/ProductDetail/Wurth-Elektronik/614004190021?qs=DvsTirrhYfPiiHUeEQNGBQ%3D%3D If you get one with bent mounting pins, then you may need to straighten them to insert them into the slots.

Generally I would suggest to get the best USB-A connector you can, it will improve the experience quite a bit.

**Comment by casandra9** [comment]
I ordered everything from mouser.ca. If you want I can share my BOM.
For the oscillators I used from mouser (25ppm)
[520-3225MVLC-240CNTR]
ECS-3225MVLC-240-CN-TR
24.000 MHz Multi-Vol

and

[520-3225MVLC-260CNTR]
ECS-3225MVLC-260-CN-TR
26.000 MHz Multi-Vol

and I ordered an extra set from LCSC (10ppm!)

LCSC#: [C4353708](https://www.lcsc.com/product-detail/Oscillators_YXC-OT322526MJBA4SL_C4353708.html)
Mfr.#: OT322526MJBA4SL
Mfr: YXC
SMD3225-4P Oscillators ROHS 

and

LCSC#: [C717686](https://www.lcsc.com/product-detail/Oscillators_YXC-OT322524MJBA4SL_C717686.html)
Mfr.#: OT322524MJBA4SL
Mfr: YXC
1.8V~3.3V ±10ppm 24MHz SMD3225-4P Oscillators ROHS

Shipment was super fast and super cost effective, component price was ~50ct, lol

I built two units, one with mouser osci and one with LCSC and I get 47-49MB/s on speed test on a Linux machine and 46-47MB/s on windows, connected to front USB-3 ports.

For the USB-A I used mouser 
[523-UE27AC5410H]
UE27AC5410H
Amphenol USB Connectors
CA HS:[8536690020](tel:8536690020) ECCN:EAR99 COO:CN

for USB-C mouser

[640-USB4800-03-A]
USB4800-03-A
Global Conn Tech (GCT) USB Connectors
CA HS:[8536690020](tel:8536690020) ECCN:EAR99 COO:CN

Both fit perfectly.

Thank you Alex for this wonderful project! Any obligations if I would share this on diyAudio board?

**Reply by simntonic** [reply]
Thanks for sharing and thank you Alex for that amazing project! :)

Replacing those oscillators with crystals is not a simple part replacement, correct? I would assume the routing needs to change slightly (and maybe add caps?) and the code as well since it will have the run the oscillator circuit.

I think people would really appreciate to have (almost) ready-to-order files to avoid having to solder small SMD parts. I'm going to try to change the design a bit to make it a JLCPCB order-able project :) 

Simon

**Reply by simntonic** [reply]
Oh my bad these actually are not simple crystals but proper oscillators just like the DSC6101CI2A-02x.0000 parts :)

**Reply by ataradov** [reply]
Yes, you would need to change the routing and add caps. Oscillators are so much easier to use that is is not really worth it to replace them. Quality crystals are expensive.

**Reply by simntonic** [reply]
Yes that makes sense, thanks. But those suggested parts OT32252xMJBA4SL are in fact oscillators and are drop-in replacements, correct?

**Reply by ataradov** [reply]
Yes, they are.

**Reply by simntonic** [reply]
Thanks :) And thanks again for that awesome project. Making that JLC ready order is a bit difficult since they do not have the FPGA or the USB phy. So I'll simply manually assemble one. I'll probably come back with some questions ;) 

Simon

**Comment by ataradov** [comment]
Share as you wish.

Those YXC oscillators are good and seem to be reliably available. I will be switching to them as a default part number in the future designs.

**Reply by casandra9** [reply]
I second this, very reliable and not a critical part as long as you stay within 25ppm.
JLC now has an affordable 3D printing service for the case.
